### PR TITLE
feat(op-challenger): Proposed Output Agreement Flag

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Adds a (temporary) CLI flag to the `op-challenger` to hardcode if we agree with the proposed output.

This boolean is propagated down to the game state where we provide an exposed method to check if the game state "agrees" with a given level using the proposed output agreement.

**Metadata**

Fixes CLI-4193

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6274** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6274" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6274?utm_source=stack-comment).

File: op-challenger/fault/solver.go [hunk 1]
```diff
@@ -51,50 +52,54 @@ func (s *Solver) AttemptStep(claim Claim) (StepData, error) {
 	}, nil
 }
 
-func (s *Solver) handleRoot(claim Claim) (*Claim, error) {
+func (s *Solver) handleRoot(claim Claim) (*Claim, bool, error) {
```
Comment (trianglesphere) [inline op-challenger/fault/solver.go]: Should not return a bool here. NextMove + these should take in a bool like `AgreeWithGame` or `DisagreeWithGame` - it's about agreement/disagreement with the output that this game is countering

File: op-challenger/cmd/main_test.go [hunk 2]
```diff
@@ -11,9 +12,10 @@ import (
 )
 
 var (
-	l1EthRpc         = "http://example.com:8545"
-	gameAddressValue = "0xaa00000000000000000000000000000000000000"
-	alphabetTrace    = "abcdefghijz"
+	l1EthRpc                = "http://example.com:8545"
+	gameAddressValue        = "0xaa00000000000000000000000000000000000000"
+	alphabetTrace           = "abcdefghijz"
+	agreeWithProposedOutput = "true"
```
Comment (ajsutton) [inline op-challenger/cmd/main_test.go:18]: We should add some tests for this new flag to ensure the CLI option makes it through to the config correctly.

**Review by ajsutton — APPROVED** [review]
LGTM. Just worth adding tests for the new CLI option.  The op-program L1TrustRPC tests are a good pattern to follow:
```
func TestL1TrustRPC(t *testing.T) {
	t.Run("DefaultFalse", func(t *testing.T) {
		cfg := configForArgs(t, addRequiredArgs())
		require.False(t, cfg.L1TrustRPC)
	})
	t.Run("Enabled", func(t *testing.T) {
		cfg := configForArgs(t, addRequiredArgs("--l1.trustrpc"))
		require.True(t, cfg.L1TrustRPC)
	})
	t.Run("EnabledWithArg", func(t *testing.T) {
		cfg := configForArgs(t, addRequiredArgs("--l1.trustrpc=true"))
		require.True(t, cfg.L1TrustRPC)
	})
	t.Run("Disabled", func(t *testing.T) {
		cfg := configForArgs(t, addRequiredArgs("--l1.trustrpc=false"))
		require.False(t, cfg.L1TrustRPC)
	})
}
```

File: op-challenger/cmd/main_test.go (hunk 2 above)
Comment (trianglesphere) [reply op-challenger/cmd/main_test.go:18]: added.

**Pull request merged**